sphinx/themes/basic/static/searchtools.js

```diff
@@ -328,13 +328,14 @@ const Search = {
     for (const [title, foundTitles] of Object.entries(allTitles)) {
       if (title.toLowerCase().trim().includes(queryLower) && (queryLower.length >= title.length/2)) {
         for (const [file, id] of foundTitles) {
-          let score = Math.round(100 * queryLower.length / title.length)
+          let score = Math.round(Scorer.title * queryLower.length / title.length);
+          let boost = titles[file] === title ? 1 : 0;  // add a boost for document titles
```

Comment:
On second look, could these be declared as `const`?

```suggestion
          const score = Math.round(Scorer.title * queryLower.length / title.length);
          const boost = titles[file] === title ? 1 : 0;  // add a small boost for document titles
```

Reply:
I thinkg it's better using a const as well. But on a second thought, I'm wondering whether a +1 is sufficient.

Reply:
Previously a title and subsection title with the same text would have equal scores, leaving their relative ranking undefined.

Any positive value here should have the effect of elevating the main-document titles above same-named subsection titles in the search results.

A single-integer increment is used because ideally we don't want the main document titles to move up in the rankings 'too much' and overtake other matches.  That is possible, though, especially given that some scores are fractional.  So I have the opposite worry: that +1 might be too much.

Reply:
(a good way to figure these out could be to develop counterexamples and add test cases for them)